amoro-ams/src/main/java/org/apache/amoro/server/dashboard/DashboardServer.java

```diff
@@ -355,7 +354,8 @@ private EndpointGroup apiGroup() {
 
   public void preHandleRequest(Context ctx) {
     String uriPath = ctx.path();
-    if (needApiKeyCheck(uriPath)) {
+    String requestSource = ctx.header("X-Request-Source");
+    if (needApiKeyCheck(uriPath) && !"frontend".equals(requestSource)) {
```

Comment:
Is `Web` a proper name for requests from the front end?

And we may better declare a static final variable for it, and the same to `X-Request-Source`